src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,296 @@
+                return null;
+            }
+
+            // cannot throw an exception here
+            return subscription
+                .SelectCatchAsync(
+                    async (value, token) =>
+                    {
+                        // duplicate context to prevent multiple event streams from sharing the same context,
+                        // clear errors/metrics/extensions, and free array pool leased arrays
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
+
+                        return await ProcessDataAsync(childContext, node, value).ConfigureAwait(false);
+                    },
+                    async (exception, token) =>
+                    {
+                        using var childContext = new ExecutionContext(context)
+                        {
+                            Errors = new ExecutionErrors(),
+                            OutputExtensions = new Dictionary<string, object?>(),
+                            Metrics = Instrumentation.Metrics.None,
+                            CancellationToken = token,
+                        };
+
+                        return await ProcessErrorAsync(context, node, exception).ConfigureAwait(false);
+                    });
+        }
+
+        /// <summary>
+        /// Processes data from the event source via <see cref="IObserver{T}.OnNext(T)"/> and
+        /// returns an <see cref="ExecutionResult"/>.
+        /// <br/><br/>
+        /// Override this method to mutate <see cref="ExecutionContext"/> as necessary, such
+        /// as changing the <see cref="ExecutionContext.RequestServices"/> property to a scoped instance.
+        /// </summary>
+        protected virtual async Task<ExecutionResult> ProcessDataAsync(ExecutionContext context, ExecutionNode node, object? value)
+        {
+            var result = new ExecutionResult(context);
+
+            try
+            {
+                var executionNode = BuildSubscriptionExecutionNode(node.Parent!, node.GraphType!, node.Field, node.FieldDefinition, node.IndexInParentNode, value!);
+
+                if (context.Listeners != null)
+                {
+                    foreach (var listener in context.Listeners)
+                    {
+                        await listener.BeforeExecutionAsync(context)
+                            .ConfigureAwait(false);
+                    }
+                }
+
+                // Execute the whole execution tree and return the result
+                await ExecuteNodeTreeAsync(context, executionNode).ConfigureAwait(false);
```

Comment:
See code from "normal" execution in `DocumentExecuter`:
```
 if (context.Errors.Count > 0)
                {
                    return new ExecutionResult
                    {
                        Errors = context.Errors,
                        Perf = metrics.Finish()
                    };
                }

                executionOccurred = true;
```

If any listener adds error into context then code returns **non-executed** ExecutionResult with errors. In contrast here I see that you always execute node and add errors from context on line 218 at the very end: `result.AddErrors(context.Errors);`

Why so?

Reply:
I'll look.  I thought I had tests for this stuff, but it should align with the document executer, and if it doesn't match, it should probably be changed.

Reply:
I see.  I will fix and add a test to verify that if a document listener adds an error before execution, it will be returned like a validation error, before executing the document.

Reply:
Also fixed so throwing an error within a document listener's 'after' event will not remove the result (matching that of the document executer).  Added tests

Reply:
Just FYI, the regular document executer receives an ExecutionResult directly from the ExecutionStrategy.  Here I was able to clean up the code a bit so it doesn't have multiple locations in the code where it creates the ExecutionResult.  Seemed to lay out a lot nicer.

Reply:
ok